pkg/apis/submariner.io/v1/types.go

```diff
@@ -175,3 +175,97 @@ func (c *Connection) SetStatus(status ConnectionStatus, messageFormat string, a
 	c.Status = status
 	c.StatusMessage = fmt.Sprintf(messageFormat, a...)
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type GlobalEgressIP struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the specification of the desired behavior.
+	Spec GlobalEgressIPSpec `json:"spec"`
+
+	// The most recently observed status. Read-only.
+	// +optional
+	Status GlobalEgressIPStatus `json:"status,omitempty"`
+}
+
+type GlobalIPAllocationSpec struct {
+	// The requested number of contiguous GlobalIPs to allocate from the Globalnet CIDR assigned to the cluster.
+	// If not specified, defaults to 1.
+	// +optional
+	NumGlobalIPs *int `json:"numGlobalIPs,omitempty"`
+}
+
+type GlobalEgressIPSpec struct {
+	// Specifies how GlobalIPs should be allocated.
+	// +optional
+	Allocation GlobalIPAllocationSpec `json:"allocation,omitempty"`
```

Comment:
@tpantelis why not directly use `NumGlobalIPs` here instead of additional `GlobalIPAllocationSpec`?

Reply:
For reuse - it avoids defining `NumGlobalIPs` and its semantics twice and also if we add `GlobalIP` or something else. If you think it's overkill, we can remove it.

Reply:
Even if we add `GlobalIP` or something else in the future, it is also possible that we might add it only to one of them (f.e, we might allow admin to specify GlobalIP as part of `ClusterGlobalEgressIP` but not for regular users in `GlobalEgressIP` - just a thought). So, maybe we can remove the Allocation for now.